ftests/nuxeo-server-tests/src/test/java/org/nuxeo/ftest/server/ITNuxeoIdempotentRequestTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.ftest.server;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.nuxeo.ecm.platform.web.common.idempotency.NuxeoIdempotentFilter.HEADER_KEY;
+import static org.nuxeo.functionaltests.AbstractTest.NUXEO_URL;
+import static org.nuxeo.functionaltests.Constants.ADMINISTRATOR;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+import org.nuxeo.client.NuxeoClient;
+import org.nuxeo.client.objects.Document;
+import org.nuxeo.ecm.core.query.sql.NXQL;
+import org.nuxeo.ecm.platform.dublincore.constants.DublinCoreConstants;
+import org.nuxeo.functionaltests.RestHelper.NuxeoClientForNuxeo;
+
+/**
+ * Tests for idempotent request mechanism.
+ *
+ * @since 11.5
+ */
+public class ITNuxeoIdempotentRequestTest {
+
+    private static final String TEST_KEY = "idempotenttestkey" + new Date().getTime();
+
+    private static final NuxeoClient.Builder CLIENT_BUILDER = new NuxeoClientForNuxeo.BuilderForNuxeo().url(
+            NUXEO_URL).authentication(ADMINISTRATOR, ADMINISTRATOR).schemas("*");
+
+    private static final NuxeoClient CLIENT = CLIENT_BUILDER.connect();
+
+    private static final NuxeoClient IDEMPOTENT_CLIENT = CLIENT_BUILDER.header(HEADER_KEY, TEST_KEY).connect();
+
+    private static final String PARENT_PATH = "/default-domain/workspaces/";
+
+    private static final String QUERY_CHILDREN = String.format("SELECT * FROM Document WHERE %s STARTSWITH '%s'",
+            NXQL.ECM_PATH, PARENT_PATH);
+
+    private static final String TEST_TITLE = "testdoc";
+
+    private static final String TEST_TYPE = "File";
+
+    protected String createDocument(NuxeoClient client) {
+        Document document = Document.createWithName(TEST_TITLE, TEST_TYPE);
+        document.setProperties(Map.of(DublinCoreConstants.DUBLINCORE_TITLE_PROPERTY, TEST_TITLE));
+        Document created = client.repository().createDocumentByPath(PARENT_PATH, document);
+        waitForAsyncWork();
+        return created.getId();
+    }
+
+    /**
+     * Prevents from random failures when counting children.
```

Comment:
How come? There should be no async jobs involved here, and I guess the transaction should always have been committed when getting  the answer from `createDocument` @kevinleturc?

Reply:
The rest query API (used to count children) relies on the page provider REST_API_SEARCH_ADAPTER, using ElasticSearch and not the CoreSession, hence the need for this to avoid random failures.

Note i needed this trick in Explorer ftests too, so i think this whole method would be useful on the RestHelper API.

Reply:
And just to answer the fact that this would be needed to get the document id after creation: it's not.
I'm just following a test pattern where the async jobs should be waited for after the action triggering them, as shutting down the test server while async jobs are still running is also exposing tests to random failures, with errors in logs, typically).